core/src/main/java/software/amazon/awssdk/SdkRequest.java

```diff
@@ -15,12 +15,63 @@
 
 package software.amazon.awssdk;
 
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+import java.util.Optional;
+
 /**
  * The base class for all SDK requests.
  *
- * TODO: SDK-specific options on the {@link AmazonWebServiceRequest} should be migrated here as part of the base-model refactor.
- *
  * @see SdkResponse
  */
-public abstract class SdkRequest {
+public abstract class SdkRequest<B extends SdkRequest.Builder<B, R, RequestOverrideConfigT>,
+        R extends SdkRequest<B, R, RequestOverrideConfigT>,
+        RequestOverrideConfigT>
+        implements ToCopyableBuilder<B, R> {
+
+    private final RequestOverrideConfigT requestOverrideConfig;
+
+    protected SdkRequest(B builder) {
+        requestOverrideConfig = builder.requestOverrideConfig();
+    }
+
+    public Optional<RequestOverrideConfigT> requestOverrideConfig() {
+        return Optional.ofNullable(requestOverrideConfig);
+    }
```

Comment:
Could this be Optional<? extends RequestOverrideConfig> (or whatever the parent class is) here and Optional<Concrete> in the children, so that we don't have to parametrize the base with the type?

Reply:
+1 sounds good.